raiden-ts/src/transfers/epics/locked.ts

```diff
@@ -477,6 +492,304 @@ function makeAndSignWithdrawConfirmation(
   );
 }
 
+function receiveTransferSigned(
+  state$: Observable<RaidenState>,
+  action: messageReceivedTyped<Signed<LockedTransfer>>,
+  { address, log, network, signer, config$ }: RaidenEpicDeps,
+): Observable<
+  | transferSigned
+  | transfer.failure
+  | transferProcessed
+  | transferSecretRequest
+  | matrixPresence.request
+> {
+  const secrethash = action.payload.message.lock.secrethash;
+  const meta = { secrethash, direction: Direction.RECEIVED };
+  return combineLatest([state$, config$]).pipe(
+    first(),
+    mergeMap(([state, { revealTimeout, caps }]) => {
+      const transfer: Signed<LockedTransfer> = action.payload.message;
+      if (secrethash in state.received) {
+        log.warn('transfer already present', action.meta);
+        const msgId = transfer.message_identifier;
+        // if transfer matches the stored one, re-send Processed once
+        if (
+          state.received[secrethash].partner === action.meta.address &&
+          state.received[secrethash].transfer[1].message_identifier.eq(msgId)
+        ) {
+          // transferProcessed again will trigger messageSend.request
+          return of(
+            transferProcessed({ message: state.received[secrethash].transferProcessed![1] }, meta),
+          );
+        } else return EMPTY;
+      }
+
+      // full balance proof validation
```

Comment:
Neglectable: Maybe sounds like a good candidate for a separate function, since this one here is already quite long? Did a quick search but wasn't able to find other places where we validate the balance proof

Reply:
We do in all places handling incoming EnvelopeMessages: receiving LockedTransfer, Unlock, LockExpired & RefundTransfer. The last one we don't handle properly yet, the others have specific rules to validate (specially related to amounts), but maybe some could be extracted, I'll check, thanks.